web/src/engine/keyboard/src/keyboards/loaders/domKeyboardLoader.ts

```diff
@@ -28,54 +29,23 @@ export class DOMKeyboardLoader extends KeyboardLoaderBase {
     this.performCacheBusting = cacheBust || false;
   }
 
-  protected loadKeyboardInternal(
-    uri: string,
-    errorBuilder: KeyboardLoadErrorBuilder,
-    id?: string
-  ): Promise<Keyboard> {
-    const promise = new ManagedPromise<Keyboard>();
-
-    if(this.performCacheBusting) {
+  protected async loadKeyboardBlob(uri: string): Promise<Blob> {
+    if (this.performCacheBusting) {
       uri = this.cacheBust(uri);
     }
 
-    try {
-      const document = this.harness._jsGlobal.document;
-      const script = document.createElement('script');
-      if(id) {
-        script.id = id;
-      }
-      document.head.appendChild(script);
-      script.onerror = (err: any) => {
-        promise.reject(errorBuilder.missingError(err));
-      }
-      script.onload = () => {
-        if(this.harness.loadedKeyboard) {
-          const keyboard = this.harness.loadedKeyboard;
-          this.harness.loadedKeyboard = null;
-          promise.resolve(keyboard);
-        } else {
-          promise.reject(errorBuilder.scriptError());
-        }
-      }
-
-      // On the oldest mobile devices we support, Promise.finally may not actually exist.
-      // Fortunately... it's not that hard of an issue to work around.
-      // Note:  es6-shim doesn't polyfill Promise.finally!
-      promise.then(() => {
-        // It is safe to remove the script once it has been run (https://stackoverflow.com/a/37393041)
-        script.remove();
-      }).catch(() => {
-        script.remove();
-      });
-
-      // Now that EVERYTHING ELSE is ready, establish the link to the keyboard's script.
-      script.src = uri;
-    } catch (err) {
-      return Promise.reject(err);
+    const response = await fetch(uri);
+    if (!response.ok) {
+      throw new KeyboardMissingError(`Cannot find the keyboard at ${uri}.`, new Error(`HTTP ${response.status} ${response.statusText}`));
     }
+    return response.blob();
```

Comment:
```suggestion
    try {
      const blob = await response.blob();
      return blob;
    } catch(e) {
      throw errorBuilder.downloadKeyboardError(`Unable to retrieve blob from keyboard at ${uri}.`, e);
    }
```

Again `KeyboardMissingError` may not be the best choice, but we want to wrap our exceptions so we only pass known exceptions up the chain